src/python/pants/backend/tools/yamllint/subsystem.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript
+from pants.option.option_types import ArgsListOption, SkipOption, StrListOption, StrOption
+from pants.util.strutil import softwrap
+
+
+class Yamllint(PythonToolBase):
```

Comment:
We'll want a lockfile for this, check out some of the other Python tools for how that looks

Reply:
I've copied the approach from ClangFormat in https://github.com/pantsbuild/pants/pull/17821/commits/d4ba3ccec669b20d912c0277473b36af6d786ce4.